src/hotspot/share/gc/shenandoah/shenandoahOldGeneration.hpp

```diff
@@ -47,6 +85,61 @@ class ShenandoahOldGeneration : public ShenandoahGeneration {
     return "OLD";
   }
 
+  // See description in field declaration
+  void set_promoted_reserve(size_t new_val);
+  size_t get_promoted_reserve() const;
+
+  // The promotion reserve is increased when rebuilding the free set transfers a region to the old generation
+  void augment_promoted_reserve(size_t increment);
+
+  // This zeros out the expended promotion count after the promotion reserve is computed
+  void reset_promoted_expended();
+
+  // This is incremented when allocations are made to copy promotions into the old generation
+  size_t expend_promoted(size_t increment);
+
+  // This is used to return unused memory from a retired promotion LAB
+  size_t unexpend_promoted(size_t decrement);
```

Comment:
May be mention in the `expend_promoted` comment that the accounting expends forward the space for an entire PLAB when its allocated, but "unexpends" any unused portion thereof when the PLAB is retired.

What this tells me is that `_promoted_expended` is non-monotonic in the evacuation phase as we expend and unexpend, which means that checks against promoted reserve might also be subject to "flickering" a bit during the latter phases of evacuation.

Reply:
I've added a comment to the field these methods operate on.